python/tests/test_functions.py

```diff
@@ -1241,8 +1241,8 @@ def test_func_SLIM_VMAG(self):
             for t in range(self.T):
                 for bus in net.buses:
                     self.assertTrue(bus.has_flags('variable','voltage magnitude'))
-                    dv = np.maximum(bus.v_max-bus.v_min,eps)
-                    vmid = 0.5*(bus.v_max+bus.v_min)
+                    dv = np.maximum(bus.v_max_reg-bus.v_min_reg,eps)
+                    vmid = 0.5*(bus.v_max_reg+bus.v_min_reg)
```

Comment:
Here you are using v reg limits for SP and SLIM functions. This is not correct. We agreed on using v reg limits only of regulation constraints. From my email:

My suggestion would be just to use the limits that make the most sense for now. For regulation, "reg" limits. For the rest probably "norm" (I can't recall all functions/constraints so help me make sure I am not missing something). In the future we will add "options" to constraints and hence a user will be able to control things like constructing a constraint with "emer" instead of "norm".